### ISSUE TITLE
Still maintaned?

### ISSUE THREAD

**Issue by Blunderchips** [ngxs-labs/select-snapshot#104]
Is this project still under active development and if so is there project roadmap/plan publicly available with what features are planned?

**Comment by arturovt** [comment]
Yeah, I've published `@ngxs-labs/select-snapshot@3.0.0` which is compatible with Angular 10.0.5+. Could you give it a try?

**Comment by Blunderchips** [comment]
Hello @arturovt sure thing, anything in particliar you would like me to test out?

**Comment by jayordway** [comment]
Can this be moved out of labs and into the core version?

**Comment by dewiktor2** [comment]
> Can this be moved out of labs and into the core version?

This will be great ;)

**Comment by roldengarm** [comment]
I've noticed that @SelectSnapshot() doesn't always return the latest value and we're in the process of removing it as it's unreliable.

**Comment by markwhitfeld** [comment]
@roldengarm You may be interested in using a utility like the one I made in this playground (if you are using Ivy):
https://stackblitz.com/edit/ngxs-select-dispatch-utils-v0-bmecrf?file=src/app/app.component.ts

It doesn't use a decorator, but the syntax is actually simpler, and more type-safe.
The `selectSnapshot` util is also slightly different in that it returns a function which you invoke from the template to get the underlying value. PS. it would not trigger change detection for you when it changes, so you would need to be aware of this.
Let me know what you think?

**Comment by mchlroy** [comment]
@roldengarm Could we have more info on this if possible? 

Are the cases where it doesn't return the latest value known? We are using it across our application, and reading this makes me a bit concerned. Should we refrain from using **@SelectSnapshot()** at all? 
Will there be a replacement? 
What about **ViewSelectSnapshot()**, does this one have the same problem?

Thanks.

**Comment by roldengarm** [comment]
@mchlroy  sorry for the late reply.

I couldn't pinpoint when it didn't work as expected, just noticed that it had the wrong value a couple of times. I've removed it entirely from our app.

**Comment by jayordway** [comment]
Has this been added to the core library or is there an alternative way to leverage such functionality in the latest version of ngxs/store?

**Comment by markwhitfeld** [comment]
@jayordway I would recommend that you use the utility I linked in [this comment](https://github.com/ngxs-labs/select-snapshot/issues/104#issuecomment-1124301807):

> https://stackblitz.com/edit/ngxs-select-dispatch-utils-v0-bmecrf?file=src/app/app.component.ts

>It doesn't use a decorator, but the syntax is actually simpler, and more type-safe.
The selectSnapshot util is also slightly different in that it returns a function which you invoke from the template to get the underlying value. PS. it would not trigger change detection for you when it changes, so you would need to be aware of this.
Let me know what you think?

**Comment by markwhitfeld** [comment]
We would love feedback on this utility that I linked.
You can add it into your project and use it. Depending on your angular version, you may need to change the `ɵɵdirectiveInject` to `inject` (now exposed in Angular 14).

The issue with the decorator approach of `@SelectSnapshot` and even `@Select` from the main library is that they lack the ability to provide type safety or type inference and they use a static global in order to gain access to the store. This causes issues with MFE apps and with server side rendering in some situations.

We would advise moving away from the decorator approach to the approach I linked in the example.

See this discussion on the proposed utils:
https://github.com/ngxs/store/discussions/1923
